File: frontend/src/types/message.ts
export type MessageResponse = {
  id: string;
  interview_id: string;
  role: "assistant" | "candidate" | "system";
  content: string;
  question_number: number | null;
  difficulty_level: number | null;
  answer_quality_score: number | null;
  response_time_ms: number | null;
  paste_detected: boolean | null;
  created_at: string | null;
};

File: api/app/presentation/schemas/message.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel


class MessageResponse(BaseModel):
    id: UUID
    interview_id: UUID
    role: Literal["assistant", "candidate", "system"]
    content: str
    question_number: int | None = None
    difficulty_level: float | None = None
    answer_quality_score: float | None = None
    response_time_ms: int | None = None
    paste_detected: bool | None = None
    created_at: datetime | None = None
